File: examples/simplersble/src/bin/list_adapters.rs
use simplersble;

fn main() {

    println!(
        "Bluetooth enabled: {}",
        simplersble::Adapter::bluetooth_enabled().unwrap()
    );

    let mut adapters = simplersble::Adapter::get_adapters().unwrap();

    // If the adapter list is empty, print a message and exit
    if adapters.is_empty() {
        println!("No adapters found.");
        return;
    }

    for adapter in adapters.iter_mut() {
        println!("Adapter: {} [{}]", adapter.identifier().unwrap(), adapter.address().unwrap());
    }
}

File: examples/simpleble/cpp/list_adapters/list_adapters.cpp
#include <iostream>

#include "simpleble/Adapter.h"
#include "simpleble/Backend.h"
#include "simpleble/Utils.h"

int main() {
    std::cout << "Using SimpleBLE version: " << SimpleBLE::get_simpleble_version() << std::endl;
    std::cout << "Bluetooth enabled: " << SimpleBLE::Backend::first_bluetooth_enabled().has_value() << std::endl;

    auto adapter_list = SimpleBLE::Adapter::get_adapters();

    if (adapter_list.empty()) {
        std::cout << "No adapter found" << std::endl;
        return EXIT_FAILURE;
    }

    for (auto& adapter : adapter_list) {
        std::cout << "Adapter: " << adapter.identifier() << " [" << adapter.address() << "]" << std::endl;
    }

    return EXIT_SUCCESS;
}
